lib/src/fty_common_asset.cc

```diff
@@ -652,12 +654,12 @@ namespace fty {
 
     void ExtendedAsset::setPriority (int priority)
     {
-        priority_ = priority;
+        priority_ = static_cast<uint8_t>(priority);
     }
 
     void ExtendedAsset::setPriority (const std::string & priority)
     {
-        priority_ = priority[1] - '0';
+        priority_ = fty::convert<uint8_t>(priority);
```

Comment:
cool, why in other places is static_cast?

Reply:
Though it was only to convert from different domains (i.e. string to integer). Shall I use also for uint64_t to uint8_t?

Reply:
yes, fty::convert uses static_cast too... but we could insert check on overflow etc...